src/cloud-cli/meltano/cloud/cli/deployment.py

```diff
@@ -0,0 +1,468 @@
+        "see `meltano cloud deployment list` for the available options."
+    ),
+    prompt=True,
+)
+@pass_context
+@run_async
+async def use_deployment(  # noqa: D103
+    context: MeltanoCloudCLIContext,
+    deployment_name: str,
+) -> None:
+    deployment_name = slugify(deployment_name)
+    if context.deployments is None:  # Interactive config was not used
+        context.deployments = await _get_deployments(context.config)
+        if deployment_name not in {x["deployment_name"] for x in context.deployments}:
+            raise click.ClickException(
+                f"Unable to use deployment named {deployment_name!r} - no available "
+                "Meltano Cloud project deployment matches name.",
+            )
+    context.config.default_deployment_name = deployment_name
+    context.config.write_to_file()
+    click.secho(
+        (
+            f"Set {deployment_name!r} as the default Meltano Cloud project "
+            "deployment for future commands"
+        ),
+        fg="green",
+    )
+
+
+@deployment_group.command("create")
+@click.option(
+    "--name",
+    "deployment_name",
+    help="A name which uniquely identifies the new Meltano Cloud project deployment",
+    prompt=True,
+)
+@click.option(
+    "--environment",
+    "environment_name",
+    help="The name of the Meltano environment to be deployed",
+    prompt=True,
+)
+@click.option(
+    "--git-rev",
+    help=(
+        "The git revision this deployment should track, such as a branch name "
+        "or commit hash"
+    ),
+    prompt=True,
+)
+@pass_context
+@run_async
+async def create_deployment(
+    context: MeltanoCloudCLIContext,
+    deployment_name: str,
+    environment_name: str,
+    git_rev: str,
+) -> None:
+    """Create a new Meltano Cloud project deployment."""
+    async with DeploymentsCloudClient(config=context.config) as client:
+        if await client.deployment_exists(deployment_name=deployment_name):
+            raise click.ClickException(
+                f"Deployment {slugify(deployment_name)!r} already exists. "
+                "Use `meltano cloud deployment update` to update an "
+                "existing Meltano Cloud project deployment.",
+            )
+        with yaspin(text="Creating deployment - this may take several minutes..."):
+            deployment = await client.update_deployment(
+                deployment_name=deployment_name,
+                environment_name=environment_name,
+                git_rev=git_rev,
+            )
+        click.secho(
+            f"Created deployment {deployment['deployment_name']!r}",
+            fg="green",
+        )
+
+
+@deployment_group.command(
+    "update",
+    short_help=(
+        "Update a Meltano Cloud project deployment, using the latest commit "
+        "from the tracked git rev."
+    ),
+)
+@click.option(
+    "--name",
+    "deployment_name",
+    help="A name which uniquely identifies the new Meltano Cloud project deployment",
```

Comment:
@pnadolny13 There is no normal situation where you'd want to do a deployment with `--force`, but it's provided as an escape hatch that's useful for when things go wrong. It will probably only be used if a Meltano team member asks the user to use it.

Reply:
@WillDaSilva ah ok that makes total sense. I was confused because my code is auto synced with cloud (i.e. I think we do a fresh git clone everytime) and had forgotten that schedules arent auto synced yet. Definitely good to note in the docs.